packages/dcos-integration-test/extra/test_networking.py

```diff
@@ -17,6 +19,163 @@
 GLOBAL_PORT_POOL = iter(range(10000, 32000))
 
 
+class Container(enum.Enum):
+    POD = 'POD'
+
+
+class MarathonApp():
+    def __init__(self, container, network, host, vip):
+        if container == Container.POD:
+            self.__class__ = MarathonPod
```

Comment:
Could you kindly add a comment here to explain the idiom of switching class objects in order to avoid a factory class for Pod?

Reply:
I've decided to remove this hacky code. It was potentially dangerous. See https://github.com/dcos/dcos/pull/2042/commits/d65c8287e268d489e8b24615097675b5a939fd79